gdpr/impl_test.go

```diff
@@ -225,14 +230,26 @@ func TestAllowActivities(t *testing.T) {
 		{
 			description:      "Allow PI - known vendor with Yes GDPR",
 			bidderName:       bidderAllowedByConsent,
+			bidderCoreName:   bidderAllowedByConsent,
 			gdprDefaultValue: "1",
 			gdpr:             SignalYes,
 			consent:          vendor2AndPurpose2Consent,
 			passID:           true,
 		},
+		{
+			description:      "Allow PI - known Alias vendor GVLID with Yes GDPR",
+			bidderName:       aliasedBidderAllowedByConsent,
+			bidderCoreName:   bidderAllowedByConsent,
+			gdprDefaultValue: "1",
+			gdpr:             SignalYes,
+			consent:          vendor2AndPurpose2Consent,
+			passID:           true,
+			aliasGVLIDs:      map[string]uint16{"appnexus1": 2},
+		},
```

Comment:
How about adding another test case just below this new one that ensures the GVL for the alias is what's affecting `passID`:
```
{
	description:      "Don't allow PI - known alias vendor with Yes GDPR, alias vendor does not consent to purpose 2",
	bidderName:       aliasedBidderAllowedByConsent,
	bidderCoreName:   bidderAllowedByConsent,
	gdprDefaultValue: "1",
	gdpr:             SignalYes,
	consent:          vendor2AndPurpose2Consent,
	passID:           false,
	aliasGVLIDs:      map[string]uint16{"appnexus1": 1},
},
```
In this case the GVL ID of 1 does not exist in the fetched vendor list so legal basis will not be established resulting in a `passID` of false.

Reply:
Added the another test case